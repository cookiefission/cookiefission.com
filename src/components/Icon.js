import React from 'react';

const Icon = (props) => {
  const name = props.name || 'warning';
  const size = props.size || 'lg';
  const classes = `fa fa-${name} fa-${size}`;

  return(
    <div className="Icon-wrapper">
      <i className={classes} />
    </div>
  );
};

export default Icon;
