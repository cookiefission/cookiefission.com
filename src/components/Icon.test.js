import React from 'react';
import Icon from './Icon';
import renderer from 'react-test-renderer';

describe('<Icon />', () => {
  it('matches the snapshot', () => {
    const component = renderer.create(<Icon />);

    expect(component.toJSON()).toMatchSnapshot();
  });

  context('when an name is provided', () => {
    it('matches the snapshot', () => {
      const component = renderer.create(<Icon name='github' />);

      expect(component.toJSON()).toMatchSnapshot();
    });
  });

  context('when a size is provided', () => {
    it('matches the snapshot', () => {
      const component = renderer.create(<Icon size='lg' />);

      expect(component.toJSON()).toMatchSnapshot();
    });
  });
});
