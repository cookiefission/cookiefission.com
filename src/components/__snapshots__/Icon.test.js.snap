exports[`<Icon /> matches the snapshot 1`] = `
<div
  className="Icon-wrapper">
  <i
    className="fa fa-warning fa-lg" />
</div>
`;

exports[`<Icon /> when a size is provided matches the snapshot 1`] = `
<div
  className="Icon-wrapper">
  <i
    className="fa fa-warning fa-lg" />
</div>
`;

exports[`<Icon /> when an name is provided matches the snapshot 1`] = `
<div
  className="Icon-wrapper">
  <i
    className="fa fa-github fa-lg" />
</div>
`;
